### ISSUE TITLE
EmptyFunctionBlock for object instance of interface false positive

### ISSUE THREAD

**Issue by aconda-mercari** [detekt/detekt#4232]
## Expected Behavior

There should be no issue as it is also not fixable.

## Observed Behavior

EmptyFunctionBlock is reported by detekt

## Steps to Reproduce

```
interface SomeViewModel {
    fun onSomeFunctionA()
    fun onSomeFunctionB()
}

@Test
fun someTest() {
    // detekt error
    object : SomeViewModel {
        override fun onSomeFunctionA() {}
        override fun onSomeFunctionB() {}
    }
}
```

## Your Environment

* Version of detekt used: 1.18.1
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by BraisGabin** [comment]
Why do you think this is a regression? Does this issue appear after you update detekt?

I don't see any issue here. If you really want an empty block you can add a comment to make clear to the reader that it's intentional. That will make the rule happy.

**Comment by mtrakal** [comment]
regression of: https://github.com/detekt/detekt/issues/148?

I override interface with empty blocks as first level of abstraction.
But it's reported as EmptyFunctionBlock, it should not for overriden functions I think.

```
abstract class SimpleTextWatcher : android.text.TextWatcher {

    override fun afterTextChanged(s: Editable?) {}

    override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
}
```
and in usage I want to implement only `onTextChanged` from TextWather:
```
binding.editText.addTextChangedListener(object : SimpleTextWatcher() {
            override fun onTextChanged(...) {
                // some code here
            }
        })
```

**Comment by BraisGabin** [comment]
Ok, but when was this regression introduced? I mean, in which version it was working as expected and in which version it stopped to work. If this is a regression it should be easy to fix but we should know when this regression was introduced.

For what I see in this issue the rule is working as expected without any regression. If you want to ignore overrides you should enable it: `ignoreOverridden: true` (it is `false` by default).

**Comment by cortinico** [comment]
As @BraisGabin mentioned, this is not a regression. The issue you linked @mtrakal is also really old and hasn't been addressed in the latest release.

**Comment by BraisGabin** [comment]
> For what I see in this issue the rule is working as expected without any regression. If you want to ignore overrides you should enable it: `ignoreOverridden: true` (it is `false` by default).

I'm closing this issue because of this ☝️. If I'm missing something here please, don't hestiate to comment and we will reconsider this issue.
